### PR TITLE
Build in Jenkins should publish linagora/tmail-lsc docker image

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM openjdk:17-jdk-alpine
```
Comment (Arsnael) [inline Dockerfile:1]: Is the lsc code really compatible with Java17?

File: Jenkinsfile [hunk 2]
```diff
@@ -17,5 +17,21 @@ pipeline {
                 }
             }
         }
+        stage('Deliver Docker image') {
```
Comment (Arsnael) [inline Jenkinsfile:20]: I didn't change all builds yet but please check the java version you need for compiling and running tests with maven above.

Then you can change 

```
agent any
```
to 
```
agent {
  label 'jdk17'
}
```
or jdk11 depending your needs. You can leave any if your build is able to work on both though

Comment (Arsnael) [reply Jenkinsfile:20]: The failed build for example ran on a jdk11 worker

Comment (quantranhong1999) [reply Jenkinsfile:20]: This should work on the jdk11 workers. Seems the worker can not download the LSC dependency, will look tmr.

File: Dockerfile (hunk 1 above)
Comment (chibenwa) [inline Dockerfile:1]: Isn't JRE enough BTW ?

Comment (chibenwa) [reply Dockerfile:1]: IMO using a JRE 17 with older code should be allright,

File: Jenkinsfile (hunk 2 above)
Comment (chibenwa) [reply Jenkinsfile:22]: Agree the build uses lower JDK versions but we can choose a higher JRE without troubbles.

**Comment by quantranhong1999** [comment]
Tested the image locally:
```bash
 ✘ hp@hp-quanth  ~/IdeaProjects/tmail-lsc   build-tmail-lsc-image ±  docker run -it -v ${PWD}/sample/ldap-to-tmail-contact/logback.xml:/opt/lsc/conf/logback.xml -v ${PWD}/sample/ldap-to-tmail-contact/lsc.xml:/opt/lsc/conf/lsc.xml linagora/tmail-lsc:latest
# JAVA_OPTS=$JAVA_OPTS ./lsc --config $CONF_DIR --synchronize all --threads 1
06:34:33,972 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
06:34:33,972 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.groovy]
06:34:33,972 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [jar:file:/opt/lsc/lib/lsc-core-2.1.6.jar!/logback.xml]
06:34:33,984 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@4ec6a292 - URL [jar:file:/opt/lsc/lib/lsc-core-2.1.6.jar!/logback.xml] is not of type file
06:34:34,018 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
06:34:34,021 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
06:34:34,027 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [CONSOLE]
06:34:34,048 |-WARN in ch.qos.logback.core.joran.util.beans.BeanDescriptionFactory@1b40d5f0 - Class 'org.lsc.utils.output.LdifLayout' contains multiple setters for the same property 'logOperations'.
06:34:34,080 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.reflections] to WARN
06:34:34,080 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting additivity of logger [org.reflections] to false
06:34:34,081 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [CONSOLE] to Logger[org.reflections]
06:34:34,082 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.lsc] to INFO
06:34:34,082 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting additivity of logger [org.lsc] to false
06:34:34,082 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [CONSOLE] to Logger[org.lsc]
06:34:34,082 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache] to WARN
06:34:34,082 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting additivity of logger [org.apache] to false
06:34:34,082 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [CONSOLE] to Logger[org.apache]
06:34:34,082 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
06:34:34,082 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [CONSOLE] to Logger[ROOT]
06:34:34,083 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
06:34:34,084 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@ea4a92b - Registering current configuration as safe fallback point

May 25 06:34:34 - INFO  - [Configuration.java : 512] - Logging configuration successfully loaded from /opt/lsc/conf/logback.xml 
May 25 06:34:34 - INFO  - [Configuration.java : 532] - LSC configuration successfully loaded from /opt/lsc/conf/
May 25 06:34:34 - DEBUG - [TMailContactDstService.java : 51] - Task bean is: org.lsc.beans.SimpleBean
May 25 06:34:34 - INFO  - [JndiServices.java : 290] - Connecting to LDAP server ldap://51.91.141.10:389/dc=parisnanterre,dc=fr as cn=openpaas,ou=accounts,dc=parisnanterre,dc=fr
May 25 06:36:45 - ERROR - [JndiServices.java : 489] - Error opening LDAP connection "openldap" to ldap://51.91.141.10:389/dc=parisnanterre,dc=fr (javax.naming.CommunicationException: 51.91.141.10:389 [Root exception is java.net.ConnectException: Connection timed out (Connection timed out)])
May 25 06:36:45 - ERROR - [Launcher.java : 229] - org.lsc.exception.LscConfigurationException: Configuration exception: javax.naming.CommunicationException: 51.91.141.10:389 [Root exception is java.net.ConnectException: Connection timed out (Connection timed out)]
May 25 06:36:45 - DEBUG - [Launcher.java : 230] - org.lsc.exception.LscConfigurationException: Configuration exception: javax.naming.CommunicationException: 51.91.141.10:389 [Root exception is java.net.ConnectException: Connection timed out (Connection timed out)]
org.lsc.exception.LscConfigurationException: Configuration exception: javax.naming.CommunicationException: 51.91.141.10:389 [Root exception is java.net.ConnectException: Connection timed out (Connection timed out)]
```

This LDAP is likely not able to be connected from public. The image seems to work well anyway. 

Need this to be merged: https://github.com/linagora/tmail-lsc/pull/6

File: Dockerfile [hunk 3]
```diff
@@ -1,6 +1,4 @@
-FROM openjdk:17-jdk-alpine
-RUN apk update && apk upgrade
-RUN apk add bash
+FROM eclipse-temurin:8-jre
```
Comment (Arsnael) [inline Dockerfile:1]: Why you regressed to java8?

Comment (quantranhong1999) [reply Dockerfile:1]: Nothing special TBH

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**